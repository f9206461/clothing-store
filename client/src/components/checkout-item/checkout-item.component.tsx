import React from "react";
import { connect } from "react-redux";

import { clearItemFromCart, addItem, removeItem } from "../../redux/cart/cart.actions";
import { CheckoutItemCont, ImgCont, ImgBox, TextCont, QtyCont, ArrowCont, ValueCont, RemoveBtnCont } from "./checkout-item.styles";

interface Props {
    cartItem: CartItem;
    clearItem: (item: CartItem) => void;
    addItem: (item: CartItem) => void;
    removeItem: (item: CartItem) => void;
}

export const CheckoutItem = ({ cartItem, clearItem, addItem, removeItem }: Props) => {
    const { name, price, quantity, imageUrl } = cartItem;

    return (
        <CheckoutItemCont>
            <ImgCont>
                <ImgBox src={imageUrl} alt="item" />
            </ImgCont>
            <TextCont>{name}</TextCont>
            <QtyCont>
                <ArrowCont onClick={() => removeItem(cartItem)}>
                    &#10094;
                </ArrowCont>
                <ValueCont>
                    {quantity}
                </ValueCont>
                <ArrowCont onClick={() => addItem(cartItem)}>
                    &#10095;
                </ArrowCont>
            </QtyCont>
            <TextCont>${price}</TextCont>
            <RemoveBtnCont onClick={() => clearItem(cartItem)}>
                &#10005;
            </RemoveBtnCont>
        </CheckoutItemCont>
    )
};

const mapDispatchToProps = (dispatch: any) => ({
    clearItem: (item: CartItem) => dispatch(clearItemFromCart(item)),
    addItem: (item: CartItem) => dispatch(addItem(item)),
    removeItem: (item: CartItem) => dispatch(removeItem(item))
})

export default connect(null, mapDispatchToProps)(CheckoutItem);