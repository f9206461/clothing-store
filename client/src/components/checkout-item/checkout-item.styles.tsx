import styled from "styled-components";

export const CheckoutItemCont = styled.div`
    width: 100%;
    display: flex;
    min-height: 100px;
    border-bottom: 1px solid darkgrey;
    padding: 15px 0;
    font-size: 20px;
    align-items: center;
`;
CheckoutItemCont.displayName = 'CheckoutItemCont';

export const ImgCont = styled.div`
    width: 23%;
    padding-right: 15px;
`;
ImgCont.displayName = 'ImgCont';

export const ImgBox = styled.img`
    width: 100%;
    height: 100%;
`;
ImgBox.displayName = 'ImgBox';

export const TextCont = styled.span`
    width: 23%;
`;
TextCont.displayName = 'TextCont';

export const QtyCont = styled.span`
    display: flex;
    width: 23%;
`;
QtyCont.displayName = 'QtyCont';

export const ArrowCont = styled.div`
    cursor: pointer;
`;
ArrowCont.displayName = 'ArrowCont';

export const ValueCont = styled.span`
    margin: 0 10px;
`;
ValueCont.displayName = 'ValueCont';

export const RemoveBtnCont = styled.div`
    padding-left: 12px;
    cursor: pointer;
`;
RemoveBtnCont.displayName = 'RemoveBtnCont';