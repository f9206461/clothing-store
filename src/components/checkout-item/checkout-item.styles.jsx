import styled from "styled-components";

export const CheckoutItemCont = styled.div`
    width: 100%;
    display: flex;
    min-height: 100px;
    border-bottom: 1px solid darkgrey;
    padding: 15px 0;
    font-size: 20px;
    align-items: center;
`;

export const ImgCont = styled.div`
    width: 23%;
    padding-right: 15px;
`;

export const ImgBox = styled.img`
    width: 100%;
    height: 100%;
`;

export const TextCont = styled.span`
    width: 23%;
`;

export const QtyCont = styled.span`
    display: flex;
    width: 23%;
`;

export const ArrowCont = styled.div`
    cursor: pointer;
`;

export const ValueCont = styled.span`
    margin: 0 10px;
`;

export const RemoveBtnCont = styled.div`
    padding-left: 12px;
    cursor: pointer;
`;