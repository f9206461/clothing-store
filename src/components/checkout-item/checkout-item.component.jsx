import React from "react";
import { connect } from "react-redux";

import { clearItemFromCart, addItem, removeItem } from "../../redux/cart/cart.actions";
import { CheckoutItemCont, ImgCont, ImgBox, TextCont, QtyCont, ArrowCont, ValueCont, RemoveBtnCont } from "./checkout-item.styles";

const CheckoutItem = ({ cartItem, clearItem, addItem, removeItem }) => {
    const { name, price, quantity, imageUrl } = cartItem;

    return (
        <CheckoutItemCont>
            <ImgCont>
                <ImgBox src={imageUrl} alt="item" />
            </ImgCont>
            <TextCont>{name}</TextCont>
            <QtyCont>
                <ArrowCont onClick={() => removeItem(cartItem)}>
                    &#10094;
                </ArrowCont>
                <ValueCont>
                    {quantity}
                </ValueCont>
                <ArrowCont onClick={() => addItem(cartItem)}>
                    &#10095;
                </ArrowCont>
            </QtyCont>
            <TextCont>${price}</TextCont>
            <RemoveBtnCont onClick={() => clearItem(cartItem)}>
                &#10005;
            </RemoveBtnCont>
        </CheckoutItemCont>
    )
};

const mapDispatchToProps = (dispatch) => ({
    clearItem: (item) => dispatch(clearItemFromCart(item)),
    addItem: item => dispatch(addItem(item)),
    removeItem: item => dispatch(removeItem(item))
})

export default connect(null, mapDispatchToProps)(CheckoutItem);