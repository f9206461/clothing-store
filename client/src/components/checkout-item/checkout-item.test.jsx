import React from "react";
import { shallow } from 'enzyme';
import { CheckoutItem } from "./checkout-item.component";

describe('checkout-item', () => {
    const mockItem = {
        name: 'jacket', 
        price: 90, 
        quantity: 2, 
        imageUrl: 'https://www.testImage.com'
    };
    const clearItem = jest.fn();
    const addItem = jest.fn(); 
    const removeItem = jest.fn();
    const mockProps = {
        cartItem: mockItem, 
        clearItem, 
        addItem,  
        removeItem
    }
    const wrapper = shallow(<CheckoutItem {...mockProps} />);

    it('should render CheckoutItem component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should call removeItem when left arrow clicked', () => {
        wrapper.find('QtyCont').childAt(0).simulate('click');
        expect(removeItem).toHaveBeenCalled();
    });

    it('should call addItem when right arrow clicked', () => {
        wrapper.find('QtyCont').childAt(2).simulate('click');
        expect(addItem).toHaveBeenCalled();
    });

    it('should call clearItem when cross is clicked', () => {
        wrapper.find('RemoveBtnCont').simulate('click');
        expect(clearItem).toHaveBeenCalled();
    })
})