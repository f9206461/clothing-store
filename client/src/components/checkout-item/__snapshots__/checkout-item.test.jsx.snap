// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkout-item should render CheckoutItem component 1`] = `
<CheckoutItemCont>
  <ImgCont>
    <ImgBox
      alt="item"
      src="https://www.testImage.com"
    />
  </ImgCont>
  <TextCont>
    jacket
  </TextCont>
  <QtyCont>
    <ArrowCont
      onClick={[Function]}
    >
      ❮
    </ArrowCont>
    <ValueCont>
      2
    </ValueCont>
    <ArrowCont
      onClick={[Function]}
    >
      ❯
    </ArrowCont>
  </QtyCont>
  <TextCont>
    $
    90
  </TextCont>
  <RemoveBtnCont
    onClick={[Function]}
  >
    ✕
  </RemoveBtnCont>
</CheckoutItemCont>
`;
